src/main/java/uk/gov/hmcts/reform/pdf/service/client/PDFServiceClient.java

```diff
@@ -43,14 +43,16 @@ public PDFServiceClient(RestOperations restTemplate, URI pdfServiceBaseUrl) {
         healthEndpoint = pdfServiceBaseUrl.resolve("/health");
     }
 
-    public byte[] generateFromHtml(byte[] template, Map<String, Object> placeholders) {
+    public byte[] generateFromHtml(String serviceAuthToken,
```

Comment:
would it be nicer if something was added to the constructor that can generate tokens? some sort of function maybe?

Reply:
DSL specific. Since we _know_ the function is `generate()`, it is up for consideration but I think as an argument makes more sense instead of some lambda.

Regardless where client is created, token can be passed when actual method is called. Token could be already present as a request header in another service, can be freshly generated in some other service.

I could be wrong, but sounds like unnecessary evil